### DISCUSSION TITLE
Live board demo

### DISCUSSION THREAD

**Discussion by sjev** [micropython/micropython#15993] [General]
Hi, 

I'm working on an [industrial controller](https://github.com/rox-automation/rox-icu) that runs MicroPython (well Circuitpython to be exact due to canbus integration). 

I think it would be a great idea to setup a live demo, just like the original [pyboard live](https://micropython.org/live/) (that one is unfortunately not operational). 

Does anybody know if  the original code for the pyboard live site is available?  

Some years have passed since then and I imagine there are better solution today to set  up something like that.  What I have in mind myself is something like this:

* rpi board with a camera and REST/websocket server with FastApi , serial & can connection to the board.
* front-end with [brython](https://brython.info)  instead of js.

As for code editing, maybe an online version of vscode?

**Comment by jonnor** [comment]
For code editing for MicroPython there is WebRepl (https://micropython.org/webrepl/) and Viper IDE (https://viper-ide.org/)

**Reply by sjev** [reply]
> For code editing for MicroPython there is WebRepl (https://micropython.org/webrepl/) and Viper IDE (https://viper-ide.org/)

Thanks, I didn't know viper yet. It looks like a great tool, I'll give it  a try.

**Comment by jonnor** [comment]
For frontend in Python there is PyScript - which actually has option to run MicroPython. https://pyscript.net/
Though in practical terms it is probably still easier to just do plain old JavaScript
